### PR TITLE
Bridge: Restart the indexer

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.
Restart the indexer implementing this discussion #884 features.

# Changelog

* update indexer db and event processing
* Change health check monitoring loop to be use with the indexer
* update process compose overlay.
# Testing

To start only the bridge (Relayer Mode):
`CELESTIA_LOG_LEVEL=FATAL CARGO_PROFILE=release CARGO_PROFILE_FLAGS=--release nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash -c "just bridge native build.setup.eth-local.celestia-local.bridge --keep-tui"`

To the Relayer and the indexer independently: 
`CELESTIA_LOG_LEVEL=FATAL CARGO_PROFILE=release CARGO_PROFILE_FLAGS=--release nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash -c "just bridge native build.setup.eth-local.celestia-local.bridge.bridge-indexer --keep-tui"`
# Outstanding issues

The Indexer-Relayer Mode is not implemented. TODO

### CODE REVIEW

**Review by andygolay — APPROVED** [review]
LGTM, tests pass for me locally and code looks good.

File: protocol-units/bridge/service/tests/relayer.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+impl TryFrom<Vec<u8>> for MockAddress {
+	type Error = AddressError;
+
+	fn try_from(vec: Vec<u8>) -> Result<Self, Self::Error> {
+		Ok(MockAddress(vec))
+	}
+}
+
+fn calculated_transfer_bridfe_id(
```
Comment (0xmovses) [inline protocol-units/bridge/service/tests/relayer.rs:57]:
```suggestion
fn calculate_bridge_transfer_id(
```

Comment (0xmovses) [reply protocol-units/bridge/service/tests/relayer.rs:57]: Update any calls to this function also.

File: protocol-units/bridge/service/tests/relayer.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+			self.complete_notifier
+				.send(Err(BridgeContractError::OnChainError("Retry.".to_string())))
+				.await
+				.unwrap();
+			return Err(BridgeContractError::OnChainError("Send Tx failed, retry".to_string()));
+		}
+
+		//verify the transfer Id
+		let calcuated_bridge_transfer_id =
+			calculated_transfer_bridfe_id(&initiator.0, &recipient.0 .0, amount, nonce);
```
Comment (0xmovses) [inline protocol-units/bridge/service/tests/relayer.rs:124]:
```suggestion
			calculate_bridge_transfer_id(&initiator.0, &recipient.0 .0, amount, nonce);
```

File: protocol-units/bridge/service/tests/relayer.rs [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+async fn initiate_bridge_transfer(
+	initiator: MockAddress,
+	recipient: MockAddress,
+	amount: Amount,
+	nonce: Nonce,
+	sender: &mut UnboundedSender<BridgeContractResult<BridgeContractEvent<MockAddress>>>,
+) -> BridgeTransferId {
+	let bridge_transfer_id =
+		calculated_transfer_bridfe_id(&initiator.0, &recipient.0, amount, nonce);
```
Comment (0xmovses) [inline protocol-units/bridge/service/tests/relayer.rs:210]:
```suggestion
		calculate_bridge_transfer_id(&initiator.0, &recipient.0, amount, nonce);
```

**Review by 0xmovses — CHANGES_REQUESTED** [review]

**Pull request merged**